packages/provider/src/providers/coinbase.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  Chain,
+  ChainId,
+  EthereumProvider,
+  EthProviderRpcError,
+  ProviderProxy,
+  RawProvider,
+  TransactionResponse,
+  TxRequestBody,
+} from '@/types'
+import { HttpProvider } from 'web3-core'
+import Web3 from 'web3/types'
+import { ChainTypes, ProviderEvents } from '@/enums'
+import {
+  connectEthProvider,
+  detectCurrentEthChain,
+  getEthExplorerAddressUrl,
+  getEthExplorerTxUrl,
+  handleEthError,
+  hexToDecimal,
+  requestAddEthChain,
+  requestSwitchEthChain,
+} from '@/helpers'
+
+declare global {
+  interface Window {
+    Web3: typeof Web3
+  }
+}
+
+export class CoinbaseProvider implements ProviderProxy {
+  readonly #provider: EthereumProvider
+  #isConnected = false
+  #web3: Web3
+  #chainId?: ChainId
+  #address?: string
+
+  constructor(provider: RawProvider) {
+    this.#web3 = new window.Web3(provider as unknown as HttpProvider)
+    this.#provider = (<unknown>this.#web3?.currentProvider) as EthereumProvider
+    this.#isConnected = false
+  }
+
```

Comment:
probably you forgot to merge main or missed this method, check phantom too:
```ts
  static get providerType(): Providers {
    return Providers.Coinbase
  }

```

Reply:
fixed

Reply:
fixed